lib/nexthop.h

```diff
@@ -134,7 +139,7 @@ struct nexthop {
 	/* Encapsulation information. */
 	enum nh_encap_type nh_encap_type;
 	union {
-		vni_t vni;
+		struct vxlan_nh_encap encap_data;
```

Comment:
why a union that is just a single struct - why not just use an instance of the struct directly?

Reply:
This is done for future use, based on the nh_encap_type we can have different data to store.